plugins/module_utils/network/sonic/config/snmp/snmp.py

```diff
@@ -0,0 +1,1405 @@
+            if conf.get('vrf') and 'tag' in conf:
+                tag_list.append(str(conf.get('vrf')))
+            if len(tag_list) == 0:
+                tag_list = None
+
+            target_dict["tag"] = tag_list
+
+            target_dict['target-params'] = target_entry_name
+            target_dict['timeout'] = conf.get('timeout')
+            port = conf.get('port')
+            if not port:
+                port = 162
+            target_dict['udp'] = {'ip': conf.get('ip'), 'port': port}
+            target_dict['ietf-snmp-ext:source-interface'] = conf.get('source_interface')
+            target_list.append(target_dict)
+
+        payload_url['target'] = target_list
+        return payload_url
+
+    def build_create_enable_target_params_payload(self, config, have, overridden_or_replaced):
+        """ Build the payload for SNMP param information based on the given host configuration
+        :rtype: A dictionary
+        :returns: The payload for SNMP target
+        """
+        payload_url = {}
+        requests = []
+        target_params_list = []
+
+        server = config.get('host')
+        have_targetentry = have.get('host')
+        have_targetentry_names = []
+        if have_targetentry and not overridden_or_replaced:
+            have_targetentry_names = [agent_entry.get('name') for agent_entry in have_targetentry]
+        for conf in server:
+            target_params_dict = {}
+            target_entry_name = ""
+            matched_host = None
+            if have_targetentry:
+                matched_host = next((x for x in have_targetentry if x.get('name') == target_entry_name), None)
+            if 'name' in conf:
+                target_entry_name = conf.get('name')
+            else:
+                target_entry_name = self.get_targetentry(have_targetentry_names)
+            if overridden_or_replaced:
+                if matched_host and 'community' in conf and 'usm' in matched_host or 'user' in conf and 'v2c' in matched_host:
+                    # delete user from matched_host before replacing it with the new community
+                    # delete community from matched_host
+                    delete_target_path = 'data/ietf-snmp:snmp/target={0}'.format(target_entry_name)
+                    target_request = {'path': delete_target_path, 'method': DELETE}
+                    requests.append(target_request)
+                    delete_target_params_path = 'data/ietf-snmp:snmp/target-params={0}'.format(target_entry_name)
+                    target_params_request = {'path': delete_target_params_path, 'method': DELETE}
+                    requests.append(target_params_request)
+            target_params_dict['name'] = target_entry_name
+            type_info = {}
+            if conf.get('user') is None:
+                if 'community' in conf:
+                    if matched_host and 'user' in matched_host:
+                        self._module.fail_json(msg="A user already exists for that host name")
```

Comment:
Please provide a little more information to the customer on what they need to do to proceed:

```suggestion
                    self._module.fail_json(msg="Incompatible configuration: Before adding 'community'
                                           configuration for this host, the existing
                                           'user' configuration must be removed.")
```